src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPumpBase.cs

```diff
@@ -0,0 +1,191 @@
+
+            config.SslKeystoreLocation = _transportBindingElement.Security.SslKeystoreLocation ?? string.Empty;
+            config.SslKeystorePassword = _transportBindingElement.Security.SslKeystorePassword ?? string.Empty;
+            config.SslKeyLocation = _transportBindingElement.Security.SslKeyLocation ?? string.Empty;
+            config.SslKeyPassword = _transportBindingElement.Security.SslKeyPassword ?? string.Empty;
+            config.SslKeyPem = _transportBindingElement.Security.SslKeyPem ?? string.Empty;
+            config.SslCertificateLocation = _transportBindingElement.Security.SslCertificateLocation ?? string.Empty;
+        }
+
+        if (_transportBindingElement.Security.SecurityMode == KafkaSecurityMode.SaslPlainText)
+        {
+            config.SecurityProtocol = SecurityProtocol.SaslPlaintext;
+            config.SaslUsername = _transportBindingElement.Security.SaslUsername ?? string.Empty;
+            config.SaslPassword = _transportBindingElement.Security.SaslPassword ?? string.Empty;
+            config.SaslMechanism = SaslMechanism.Plain;
+        }
+
+        if (_transportBindingElement.Security.SecurityMode == KafkaSecurityMode.SaslSsl)
+        {
+            config.SecurityProtocol = SecurityProtocol.SaslSsl;
+            config.SslEndpointIdentificationAlgorithm = SslEndpointIdentificationAlgorithm.Https;
+            config.SslCaLocation = _transportBindingElement.Security.SslCaLocation ?? string.Empty;
+            config.SslCaPem = _transportBindingElement.Security.SslCaPem ?? string.Empty;
+
+            config.SaslUsername = _transportBindingElement.Security.SaslUsername ?? string.Empty;
+            config.SaslPassword = _transportBindingElement.Security.SaslPassword ?? string.Empty;
+            config.SaslMechanism = SaslMechanism.Plain;
+        }
+
+        return config;
+    }
+
+    public override Task StartPumpAsync(QueueTransportContext queueTransportContext, CancellationToken token)
+    {
+        _cts = new();
+        _mres = new();
+        _mres.Reset();
+
+        Consumer = new ConsumerBuilder<Null, byte[]>(BuildConfig())
+            .SetKeyDeserializer(Deserializers.Null)
+            .SetValueDeserializer(Deserializers.ByteArray)
+            .Build();
+
+        Consumer.Subscribe(_topic);
+
+        CancellationToken ct = CancellationTokenSource
+            .CreateLinkedTokenSource(_cts.Token, token).Token;
+
+        Task.Run(async () =>
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                try
+                {
+                    var consumeResult = Consumer.Consume(ct);
+                    if (consumeResult.IsPartitionEOF)
+                    {
+                        continue;
+                    }
+
+                    _logger.LogInformation("Receiving message from kafka at {topicPartitionOffset}", consumeResult.TopicPartitionOffset);
+                    try
```

Comment:
remove the try/catch that surround `OnConsumerMessage` because these exceptions will be caught by the queue middlewares and the dispatch runtime